Tests/MoyaProviderIntegrationTests.swift

```diff
@@ -299,6 +299,30 @@ final class MoyaProviderIntegrationTests: QuickSpec {
                 }
             }
         }
+
+        // Resolves ValidationType not working with multipart upload #1590
+        describe("a provider performing a multipart upload with Alamofire validation") {
+            it("only allows status code 287") {
+                let formData = HTTPBin.createMultipartFormData()
+                let target = HTTPBin.validatedUploadMultipart(formData, nil)
+                let provider = MoyaProvider<HTTPBin>()
+                provider.request(target) { result in
```

Comment:
I would restructure this testing code a little bit. Instead of using `expect`s inside the closures, we could use `waitUntil` structure. This makes sure we don't skip this test, where with using `expect`s inside the closure we are risking the situation where this test is skipped, e.g. when closure is not executed. See tests above for examples.

Reply:
Great catch. I totally forgot about that